compiler/optimizer/demand.go

```diff
@@ -109,8 +109,16 @@ func demandForOp(op dag.Op, downstream demand.Demand) demand.Demand {
 	case *dag.CommitMetaScan, *dag.DefaultScan, *dag.Deleter, *dag.DeleteScan, *dag.LakeMetaScan:
 		return demand.None()
 	case *dag.FileScan:
-		d := demand.Union(downstream, demandForExpr(op.Filter))
-		op.Fields = demand.Fields(d)
+		d := downstream
+		if mf := op.Pushdown.MetaFilter; mf != nil {
+			filterDemand := demandForExpr(mf.Expr)
+			mf.Projection = demand.Fields(filterDemand)
+			d = demand.Union(d, filterDemand)
```

Comment:
I know this doesn't hurt but I don't think we need to merge the fields needed by the metadata filter into those needed by the overall query.

Reply:
Yes this is important clarificaiton.